api/src/pcapi/core/providers/titelive_product_add_gtl_jsondata.py

```diff
@@ -0,0 +1,231 @@
+                    ean_data_list.append(extra_data)
+                else:
+                    ean_duplicates.append(ean)
+
+        eans = set(ean_datas.keys())
+
+        matching_product_in_db_count = (
+            offers_models.Product.query.filter(offers_models.Product.idAtProviders.in_(eans))
+            .options(sa.orm.load_only(offers_models.Product.id, offers_models.Product.extraData))
+            .count()
+        )
+
+        logger.info("Total rows: %s", total_rows)
+        logger.info("Total EANs: %s", len(eans))
+        logger.info("Ignoring %s EANs present more than once in CSV extract: %s", len(ean_duplicates), ean_duplicates)
+        logger.info("Total matching products in database: %s", matching_product_in_db_count)
+
+        products_and_eans_diff_count = len(eans) - matching_product_in_db_count
+
+        if products_and_eans_diff_count != 0:
+            logger.info(
+                "%s EANs in CSV are not in the database...",
+                products_and_eans_diff_count,
+            )
+
+        total = len(eans)
+        per_page = 1000
+        nb_pages = math.ceil(total / per_page)
+
+        logger.info(
+            "%s EANs will have extraData edited by batch of %s entries and a total of %s pages",
+            total,
+            per_page,
+            nb_pages,
+        )
+
+        pages = split_in_pages(list(eans), nb_pages)
+        errors: list[dict] = []
+
+        for index, eans_in_page in enumerate(pages):
+            page_id = index + 1
+
+            products = (
+                offers_models.Product.query.filter(offers_models.Product.idAtProviders.in_(eans_in_page))
+                .options(
+                    sa.orm.load_only(
+                        offers_models.Product.id, offers_models.Product.idAtProviders, offers_models.Product.extraData
+                    )
+                )
+                .all()
+            )
+
+            has_missing_entries_in_db = len(eans_in_page) - len(products) > 0
+            if has_missing_entries_in_db:
+                logger.warning(
+                    "[Page %s/%s] Partial matching: %s/%s database entries found: %s",
+                    page_id,
+                    nb_pages,
+                    len(products),
+                    len(eans_in_page),
+                    {
+                        "page_id": page_id,
+                        "ean_not_in_db_total": len(eans_in_page) - len(products),
+                        "ean_not_in_db": set(eans_in_page).difference([p.idAtProviders for p in products]),
+                    },
+                )
+
+            for product in products:
+                product.extraData[EXTRA_DATA_KEYS["ean"]] = ean_datas[EXTRA_DATA_KEYS["ean"]][EXTRA_DATA_KEYS["ean"]]
+                product.extraData[EXTRA_DATA_KEYS["code_gtl"]] = ean_datas[EXTRA_DATA_KEYS["ean"]][
+                    EXTRA_DATA_KEYS["code_gtl"]
+                ]
```

Comment:
On écrit `ean_datas[EXTRA_DATA_KEYS["ean"]]` partout ici, mais ça ne récupère pas les données de l'EAN du produit. À mon avis, il faut quelque chose comme ça : 

```suggestion
                ean = product.idAtProviders
                ean_data = ean_datas[ean]
                product.extraData[EXTRA_DATA_KEYS["ean"]] = ean_data[EXTRA_DATA_KEYS["ean"]]
                product.extraData[EXTRA_DATA_KEYS["code_clil"]] = ean_data[EXTRA_DATA_KEYS["code_clil"]]
                ...
```